algorithms/indexing/basis_vector_search/strategy.py

```diff
@@ -0,0 +1,41 @@
+"""Basis vector search strategies."""
+
+from __future__ import absolute_import, division, print_function
+
+import abc
+
+
+class Strategy(object):
+    """A base class for basis vector search strategies."""
+
+    __metaclass__ = abc.ABCMeta
```

Comment:
This is Python 2 only syntax. Do we really need metaclasses here, or would a 
```Python
def override_this_function():
    raise NotImplementedError()
```
suffice?